### PR TITLE
Visualization of GroMEts in two types (PNC and GrFN)

### PR DESCRIPTION
**Pull request by RosaRomeroGomez**
**What**

- Updated path to the GroMEt parser in package.json since now it is merged to the main branch
- Added visual encodings for parameters and variables for GroMEts (parameters are displayed with a smaller rectangle and variables in light blue). There are nodes that can be both parameters and variables
- Clean up of some visual encodings from the Feb demo within `encodings.ts`
- Added type for graph layouts.
- Clean up both views (GraphsList and ModelsList) since they don't need to display any settings.
- Ability to switch layouts (elk and dagre) for a specific gromet representation.

**Screenshots**
image
image

**Testing**

- Select the only existing model and switching between modeling frameworks and layouts. Things should work and look fine.

### CODE REVIEW

File: client/src/components/SettingsBar.vue [hunk 1]
```diff
@@ -23,7 +23,7 @@
   flex-shrink: 0;
   gap: 5px;
   justify-content: space-between;
-  min-height: calc(#{$secondary-bar-width} - 15px);
+  min-height: $secondary-bar-width;
```
Comment (YohannParis) [inline client/src/components/SettingsBar.vue:26]: ❤️

File: client/src/graphs/svg/dagre/adapter.js [hunk 2]
```diff
@@ -13,8 +13,13 @@ export default class DagreAdapter {
       if (!node.nodes) {
         node.width = this.nodeWidth;
         node.height = this.nodeHeight;
+
+        // Special case for parameters
+        if (node.data.role?.includes('parameter')) {
+          node.width = 30;
+          node.height = 30;
```
Comment (YohannParis) [inline client/src/graphs/svg/dagre/adapter.js:20]: `this.nodeWidth` & `this.nodeHeight` are defined, why not do the same thing for _parameters_?

File: client/src/graphs/svg/elk/adapter.js [hunk 3]
```diff
@@ -77,6 +77,12 @@ const injectELKOptions = (renderGraph, options) => {
     if (!node.nodes || node.nodes.length === 0) {
       node.width = node.width || options.nodeWidth;
       node.height = node.height || options.nodeHeight;
+
+      // Special case for parameters
+      if (node.data.role?.includes('parameter')) {
+        node.width = 30;
+        node.height = 30;
```
Comment (YohannParis) [inline client/src/graphs/svg/elk/adapter.js:84]: Same comment made with _Dagre_ adapter.

File: client/src/graphs/svg/encodings.ts [hunk 4]
```diff
@@ -5,13 +5,12 @@ export abstract class Colors {
...
     AGGREGATE: Colors.HIGHLIGHT,
-    ROOT_CONTAINER: '#4C566A',
-    CONTAINER: '#2E3440',
-    MODEL_VARIABLE: '#88C0D0',
-    INITIAL_CONDITION: '#81A1C1',
     OVERLAPPING: Colors.HIGHLIGHT,
+
+    // Gromet
+    CONTAINER: '#2E3440',
+    VARIABLE: '#88C0D0',
```
Comment (YohannParis) [inline client/src/graphs/svg/encodings.ts:13]: We should move all those colours to CSS cutom properties in a specific PR.

File: client/src/views/Models/components/Graphs/GlobalGraph.vue [hunk 5]
```diff
@@ -38,11 +38,16 @@
...
+    layoutChanged (): void {
+      this.refresh();
+    }
+
     mounted (): void {
        this.renderer = new EpiRenderer({
         el: this.$refs.graph,
-        adapter: new DagreAdapter(DEFAULT_RENDERING_OPTIONS),
-        renderMode: 'delta',
+        adapter: new ELKAdapter(Object.assign({}, DEFAULT_RENDERING_OPTIONS, { layout: layered })),
```
Comment (YohannParis) [inline client/src/views/Models/components/Graphs/GlobalGraph.vue:49]: If `layered` is only used by `ELKAdapter` why not importing it inside `ELKAdapter` and do the `Object.assign()` outside of this compoment.

File: client/src/views/Models/Model.vue [hunk 6]
```diff
@@ -37,10 +37,17 @@
...
               :selected-view-id="selectedViewId"
               :views="views"
+              :layouts="layouts"
+              :selected-layout-id="selectedLayoutId"
               @view-change="onSetView"
+              @layout-change="onSetLayout"
             />
           </settings-bar>
-          <global-graph v-if="selectedModel" :data="selectedGraph" @node-click="onNodeClick"/>
+          <global-graph
```
Comment (YohannParis) [inline client/src/views/Models/Model.vue:46]: to avoid loading error in case of missing graph, can we add the `v-if="selectedGraph"` back.

**Review by YohannParis — APPROVED** [review]

File: client/src/views/Models/Model.vue (hunk 6 above)
Comment (RosaRomeroGomez) [reply client/src/views/Models/Model.vue:46]: Done.

File: client/src/graphs/svg/dagre/adapter.js (hunk 2 above)
Comment (RosaRomeroGomez) [reply client/src/graphs/svg/dagre/adapter.js:20]: Done.

File: client/src/graphs/svg/elk/adapter.js (hunk 3 above)
Comment (RosaRomeroGomez) [reply client/src/graphs/svg/elk/adapter.js:84]: Done.

File: client/src/views/Models/components/Graphs/GlobalGraph.vue (hunk 5 above)
Comment (RosaRomeroGomez) [reply client/src/views/Models/components/Graphs/GlobalGraph.vue]: Done.

**Pull request merged**